packages/react-querybuilder/src/utils/formatQuery/defaultValueProcessorMongoDBByRule.ts

```diff
@@ -17,80 +17,73 @@ export const defaultValueProcessorMongoDBByRule: ValueProcessorByRule = (
-      : `{"${field}":{"$regex":"^${escapeDoubleQuotes(value)}"}}`;
+      ? `"$where":"this.${field}.startsWith(this.${value})"`
+      : `"${field}":{"$regex":"^${escapeDoubleQuotes(value)}"}`;
   } else if (operator === 'endsWith') {
     return valueIsField
-      ? `{"$where":"this.${field}.endsWith(this.${value})"}`
-      : `{"${field}":{"$regex":"${escapeDoubleQuotes(value)}$"}}`;
+      ? `"$where":"this.${field}.endsWith(this.${value})"`
+      : `"${field}":{"$regex":"${escapeDoubleQuotes(value)}$"}`;
   } else if (operator === 'doesNotContain') {
     return valueIsField
-      ? `{"$where":"!this.${field}.includes(this.${value})"}`
-      : `{"${field}":{"$not":{"$regex":"${escapeDoubleQuotes(value)}"}}}`;
+      ? `"$where":"!this.${field}.includes(this.${value})"`
+      : `"${field}":{"$not":{"$regex":"${escapeDoubleQuotes(value)}"}}`;
   } else if (operator === 'doesNotBeginWith') {
     return valueIsField
-      ? `{"$where":"!this.${field}.startsWith(this.${value})"}`
-      : `{"${field}":{"$not":{"$regex":"^${escapeDoubleQuotes(value)}"}}}`;
+      ? `"$where":"!this.${field}.startsWith(this.${value})"`
+      : `"${field}":{"$not":{"$regex":"^${escapeDoubleQuotes(value)}"}}`;
   } else if (operator === 'doesNotEndWith') {
     return valueIsField
-      ? `{"$where":"!this.${field}.endsWith(this.${value})"}`
-      : `{"${field}":{"$not":{"$regex":"${escapeDoubleQuotes(value)}$"}}}`;
+      ? `"$where":"!this.${field}.endsWith(this.${value})"`
+      : `"${field}":{"$not":{"$regex":"${escapeDoubleQuotes(value)}$"}}`;
   } else if (operator === 'null') {
-    return `{"${field}":null}`;
+    return `"${field}":null`;
   } else if (operator === 'notNull') {
-    return `{"${field}":{"$ne":null}}`;
+    return `"${field}":{"$ne":null}`;
   } else if (operator === 'in' || operator === 'notIn') {
     const valArray = toArray(value);
-    if (valArray.length > 0) {
-      return valueIsField
-        ? `{"$where":"${operator === 'notIn' ? '!' : ''}[${valArray
-            .map(val => `this.${val}`)
-            .join(',')}].includes(this.${field})"}`
-        : `{"${field}":{"${mongoOperators[operator]}":[${valArray
-            .map(val =>
-              shouldRenderAsNumber(val, parseNumbers)
-                ? `${trimIfString(val)}`
-                : `"${escapeDoubleQuotes(val)}"`
-            )
-            .join(',')}]}}`;
-    } else {
-      return '';
-    }
+    return `"${field}":{"${mongoOperators[operator]}":[${valArray
+      .map(val =>
+        shouldRenderAsNumber(val, parseNumbers)
+          ? `${trimIfString(val)}`
+          : `"${escapeDoubleQuotes(val)}"`
+      )
+      .join(',')}]}`;
   } else if (operator === 'between' || operator === 'notBetween') {
     const valArray = toArray(value);
-    if (valArray.length >= 2 && isValidValue(valArray[0]) && isValidValue(valArray[1])) {
+    if (isValidValue(valArray[0]) && isValidValue(valArray[1])) {
```

Comment:
Why was the check for minimum array length of 2 removed?

Reply:
It needs to stay. It may seem redundant, but it allows it to immediately bail on processing the rule if the array length is less than 2.